ci/travis/install-dependencies.sh

```diff
@@ -1,122 +1,173 @@
-  # Install miniconda.
-  wget -q https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh -O miniconda.sh -nv
-  bash miniconda.sh -b -p $HOME/miniconda
-  export PATH="$HOME/miniconda/bin:$PATH"
-  "${ROOT_DIR}/install-strace.sh" || true
-  pip install scipy tensorflow==$tf_version \
-    cython==0.29.0 gym \
-    opencv-python-headless pyyaml pandas==0.24.2 requests \
-    feather-format lxml openpyxl xlrd py-spy pytest pytest-timeout networkx tabulate aiohttp \
-    uvicorn dataclasses pygments werkzeug kubernetes flask grpcio pytest-sugar pytest-rerunfailures pytest-asyncio \
-    blist scikit-learn numba
-elif [[ "$PYTHON" == "3.6" ]] && [[ "$platform" == "macosx" ]]; then
-  # Install miniconda.
-  wget -q https://repo.continuum.io/miniconda/Miniconda3-4.5.4-MacOSX-x86_64.sh -O miniconda.sh -nv
-  bash miniconda.sh -b -p $HOME/miniconda
-  export PATH="$HOME/miniconda/bin:$PATH"
-  pip install scipy tensorflow==$tf_version \
-    cython==0.29.0 gym \
-    opencv-python-headless pyyaml pandas==0.24.2 requests \
-    feather-format lxml openpyxl xlrd py-spy pytest pytest-timeout networkx tabulate aiohttp \
-    uvicorn dataclasses pygments werkzeug kubernetes flask grpcio pytest-sugar pytest-rerunfailures pytest-asyncio \
-    blist scikit-learn numba
-elif [[ "$LINT" == "1" ]]; then
-  sudo apt-get update
-  sudo apt-get install -y build-essential curl unzip
-  # Install miniconda.
-  wget -q https://repo.continuum.io/miniconda/Miniconda3-4.5.4-Linux-x86_64.sh -O miniconda.sh -nv
-  bash miniconda.sh -b -p $HOME/miniconda
-  export PATH="$HOME/miniconda/bin:$PATH"
-  # Install Python linting tools.
-  pip install flake8==3.7.7 flake8-comprehensions flake8-quotes==2.0.0
-  # Install TypeScript and HTML linting tools.
-  pushd "$ROOT_DIR/../../python/ray/dashboard/client"
-    source "$HOME/.nvm/nvm.sh"
-    nvm install node
-    nvm use node
-    npm ci
-  popd
-elif [[ "$LINUX_WHEELS" == "1" ]]; then
-  sudo apt-get install docker
-  sudo usermod -a -G docker travis
-elif [[ "$MAC_WHEELS" == "1" ]]; then
-  :
-else
-  echo "Unrecognized environment."
-  exit 1
-fi
-
-# Install modules needed in all jobs.
-pip install dm-tree
-
-# Additional RLlib dependencies.
-if [[ "$RLLIB_TESTING" == "1" ]]; then
-  pip install tensorflow-probability==$tfp_version gast==0.2.2 \
-    torch==$torch_version torchvision \
-    atari_py gym[atari] lz4 smart_open
-fi
-
-# Additional streaming dependencies.
-if [[ "$RAY_CI_STREAMING_PYTHON_AFFECTED" == "1" ]]; then
-  pip install -q msgpack>=0.6.2
-fi
-
-if [[ "$PYTHON" == "3.6" ]] || [[ "$MAC_WHEELS" == "1" ]]; then
-  # Install the latest version of Node.js in order to build the dashboard.
-  source "$HOME/.nvm/nvm.sh"
-  nvm install node
-fi
-
-pip install psutil setproctitle \
-        --target="$ROOT_DIR/../../python/ray/thirdparty_files"
+{ SHELLOPTS_STACK="${SHELLOPTS_STACK-}|$(set +o); set -$-"; } 2> /dev/null  # Push caller's shell options (quietly)
```

Comment:
Push/pop are only relevant for scripts that have callers to return to (i.e., which are sourced rather than executed). Most scripts are just executed directly, but this script and `ci.sh` are sourced, hence they need to restore the state for Travis. I could turn these into functions, but there are 2 problems:

- The push/pop need to be the _first_ things that happens in the script (even surrounding the `ROOT_DIR=...` stuff) to avoid defeating their purpose.

- The push/pop in a callee will overwrite the one in a caller (unless I use different names for the same functionality, which I think might then make them confusing).

I was hoping to instead move `install-dependencies.sh` into `ci.sh` in a following PR, so that might make this easier to deal with? That way only that script will be sourced and need to have this line.